File: src/models/UserProfile.ts
export interface UserProfile {
  name?: string;
  birthday?: string;
  relationshipStage: string;
  relationshipTime: string;
  needs: string[];
  notificationTime?: string[]; // e.g. ["smart", "20:00"]
  notificationsEnabled?: boolean;
  completedCategoryIds?: string[]; // Persistent history of completed categories
}

File: lib/models/user_profile.dart
class UserProfile {
  final String relationshipStage;
  final String relationshipTime;
  final List<String> needs;

  UserProfile({
    required this.relationshipStage,
    required this.relationshipTime,
    required this.needs,
  });

  Map<String, dynamic> toJson() {
    return {
      'relationshipStage': relationshipStage,
      'relationshipTime': relationshipTime,
      'needs': needs,
    };
  }

  factory UserProfile.fromJson(Map<String, dynamic> json) {
    return UserProfile(
      relationshipStage: json['relationshipStage'] as String,
      relationshipTime: json['relationshipTime'] as String,
      needs: List<String>.from(json['needs']),
    );
  }
}
